diff -Naur nginx-1.21.5/debian/control nginx-geoip2/debian/control
--- nginx-1.21.5/debian/control 2022-01-03 02:23:30.000000000 +0000
+++ nginx-geoip2/debian/control 2022-02-21 19:31:27.886827170 +0000
@@ -13,6 +13,7 @@
                libxml2-dev,
                libgd-dev,
                libgeoip-dev,
+               libmaxminddb-dev,
                libhiredis-dev,
                liblua5.1-0-dev [!i386 !amd64 !kfreebsd-i386 !armel !armhf !arm64 !powerpc !powerpcspe !mips !mipsel],
                libluajit-5.1-dev [i386 amd64 kfreebsd-i386 armel armhf arm64 powerpc powerpcspe mips mipsel],
@@ -111,6 +112,7 @@
          libnginx-mod-http-dav-ext (= ${binary:Version}),
          libnginx-mod-http-echo (= ${binary:Version}),
          libnginx-mod-http-geoip (= ${binary:Version}),
+         libnginx-mod-http-geoip2 (= ${binary:Version}),
          libnginx-mod-http-image-filter (= ${binary:Version}),
          libnginx-mod-http-subs-filter (= ${binary:Version}),
          libnginx-mod-http-upstream-fair (= ${binary:Version}),
@@ -186,6 +188,7 @@
          libnginx-mod-http-redis2 (= ${binary:Version}),
          libnginx-mod-http-fancyindex (= ${binary:Version}),
          libnginx-mod-http-geoip (= ${binary:Version}),
+         libnginx-mod-http-geoip2 (= ${binary:Version}),
          libnginx-mod-http-headers-more-filter (= ${binary:Version}),
          libnginx-mod-http-image-filter (= ${binary:Version}),
          libnginx-mod-http-lua (= ${binary:Version}),
@@ -243,6 +246,14 @@
  Those variables include country, region, city, latitude, longitude, postal
  code, etc.

+Package: libnginx-mod-http-geoip2
+Architecture: any
+Depends: ${misc:Depends}, ${shlibs:Depends}, libmaxminddb0
+Description: GeoIP HTTP module for Nginx
+ The ngx_http_geoip2 module is a third-party plugin which creates variables
+ with values from the maxmind geoip2 databases based on the client IP
+ (default) or from a specific variable (supports both IPv4 and IPv6).
+
 Package: libnginx-mod-http-image-filter
 Architecture: any
 Depends: ${misc:Depends}, ${shlibs:Depends}
diff -Naur nginx-1.21.5/debian/copyright nginx-geoip2/debian/copyright
--- nginx-1.21.5/debian/copyright       2020-05-14 03:08:49.000000000 +0000
+++ nginx-geoip2/debian/copyright       2022-02-21 19:28:45.213976415 +0000
@@ -206,3 +206,7 @@
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
+
+Files: debian/modules/http-geoip2/*
+Copyright: Copyright (c) 2014, Lee Valentine <lee@leev.net>
+License: BSD-2-clause
diff -Naur nginx-1.21.5/debian/rules nginx-geoip2/debian/rules
--- nginx-1.21.5/debian/rules   2021-05-26 22:47:00.000000000 +0000
+++ nginx-geoip2/debian/rules   2022-02-21 19:32:01.506309702 +0000
@@ -19,6 +19,7 @@
        http-redis2 \
        http-fancyindex \
        http-geoip \
+       http-geoip2 \
        http-headers-more-filter \
        http-image-filter \
        http-lua \
@@ -133,7 +134,8 @@
                        --add-dynamic-module=$(MODULESDIR)/http-dav-ext \
                        --add-dynamic-module=$(MODULESDIR)/http-echo \
                        --add-dynamic-module=$(MODULESDIR)/http-upstream-fair \
-                       --add-dynamic-module=$(MODULESDIR)/http-subs-filter
+                       --add-dynamic-module=$(MODULESDIR)/http-subs-filter \
+                       --add-dynamic-module=$(MODULESDIR)/http-geoip2

 extras_configure_flags := \
                        $(common_configure_flags) \
@@ -171,7 +173,8 @@
                        --add-dynamic-module=$(MODULESDIR)/http-uploadprogress \
                        --add-dynamic-module=$(MODULESDIR)/http-upstream-fair \
                        --add-dynamic-module=$(MODULESDIR)/http-subs-filter \
-                       --add-dynamic-module=$(MODULESDIR)/http-vhost-traffic-status
+                       --add-dynamic-module=$(MODULESDIR)/http-vhost-traffic-status \
+                       --add-dynamic-module=$(MODULESDIR)/http-geoip2

 %:
        dh $@ --without autoreconf
